src/sql/src/plan/statement.rs

```diff
@@ -428,6 +431,75 @@ fn handle_alter_object_rename(
     })
 }
 
+fn handle_alter_index_options(
+    scx: &StatementContext,
+    AlterIndexOptionsStatement {
+        index_name,
+        if_exists,
+        options,
+    }: AlterIndexOptionsStatement,
+) -> Result<Plan, anyhow::Error> {
+    let alter_index = match scx.resolve_item(index_name) {
+        Ok(name) => {
+            let entry = scx.catalog.get_item(&name);
+            if entry.item_type() != CatalogItemType::Index {
+                bail!("{} is a {} not a index", name, entry.item_type())
+            }
+
+            let logical_compaction_window = match options {
+                AlterIndexOptionsList::Reset(o) => {
+                    let mut options: HashSet<_> =
+                        o.iter().map(|x| normalize::ident(x.clone())).collect();
+                    // Follow Postgres and don't complain if unknown parameters
+                    // are passed into ALTER INDEX ... RESET
+                    if options.remove("logical_compaction_window") {
+                        Some(LogicalCompactionWindow::Default)
+                    } else {
+                        None
+                    }
+                }
+                AlterIndexOptionsList::Set(o) => {
+                    let mut options = normalize::options(&o);
+
+                    let logical_compaction_window = match options
+                        .remove("logical_compaction_window")
+                    {
+                        Some(Value::String(window)) => match window.as_str() {
+                            "off" => Some(LogicalCompactionWindow::Off),
+                            s => Some(LogicalCompactionWindow::Custom(parse_duration::parse(s)?)),
+                        },
+                        Some(_) => bail!("\"logical_compaction_window\" must be a string"),
+                        None => None,
+                    };
+
+                    if !options.is_empty() {
+                        bail!("unrecognized parameter: \"{}\". Only \"logical_compaction_window\" is currently supported.",
+                              options.keys().next().expect("known to exist"))
+                    }
+
+                    logical_compaction_window
+                }
+            };
+
+            if let Some(logical_compaction_window) = logical_compaction_window {
+                Some(AlterIndexLogicalCompactionWindow {
+                    index: entry.id(),
+                    logical_compaction_window,
+                })
+            } else {
+                None
+            }
+        }
+        Err(_) if if_exists => {
+            // TODO(rkhaitan): better message indicating that the index does not exist.
```

Comment:
fwiw, I believe that letting this fall through and generate the `ALTER INDEX` response in the coordinator is all that's needed to provide PG-like behavior.

Reply:
No I think PG is more helpful than that

```
Test=# ALTER INDEX dummy set(parameter=1);
ERROR:  relation "dummy" does not exist
```

tbh I copied this todo because I noticed that the ALTER INDEX RENAME code had it :)

Reply:
Ah, derp! :)